source/tests/traits_tests.cpp

```diff
@@ -414,7 +418,11 @@ struct validate
         }
     }
 
+#if defined(TIMEMORY_WINDOWS)
+    template <size_t Idx, size_t... Tail, tim::enable_if_t<(sizeof...(Tail) > 0)> = 0>
```

Comment:
You can just use `std::enable_if_t` here and get rid of the pre-processor blocks.

Reply:
Unfortunately, when I changed tim::enable_if to std::enable_if I got different errors that I cannot understand or troubleshoot. The errors using std::enable_if are:

```
Build started...
1>------ Build started: Project: traits_tests, Configuration: Release x64 ------
1>traits_tests.cpp
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(400,36): error C2672: 'operator __surrogate_func': no matching overloaded function found
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(400): message : while compiling class template member function 'validate<Predicate,tim::cereal::PrettyJSONOutputArchive,tim::cereal::MinimalJSONOutputArchive,tim::cereal::XMLOutputArchive,tim::cereal::PortableBinaryOutputArchive,tim::cereal::BinaryOutputArchive>::validate(bool)'
1>        with
1>        [
1>            Predicate=tim::concepts::is_output_archive
1>        ]
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(438): message : see reference to function template instantiation 'validate<Predicate,tim::cereal::PrettyJSONOutputArchive,tim::cereal::MinimalJSONOutputArchive,tim::cereal::XMLOutputArchive,tim::cereal::PortableBinaryOutputArchive,tim::cereal::BinaryOutputArchive>::validate(bool)' being compiled
1>        with
1>        [
1>            Predicate=tim::concepts::is_output_archive
1>        ]
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(436): message : see reference to class template instantiation 'validate<Predicate,tim::cereal::PrettyJSONOutputArchive,tim::cereal::MinimalJSONOutputArchive,tim::cereal::XMLOutputArchive,tim::cereal::PortableBinaryOutputArchive,tim::cereal::BinaryOutputArchive>' being compiled
1>        with
1>        [
1>            Predicate=tim::concepts::is_output_archive
1>        ]
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(527): message : see reference to class template instantiation 'validate<tim::concepts::is_output_archive,output_archives>' being compiled
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(400,1): error C2783: 'auto validate<Predicate,tim::cereal::PrettyJSONOutputArchive,tim::cereal::MinimalJSONOutputArchive,tim::cereal::XMLOutputArchive,tim::cereal::PortableBinaryOutputArchive,tim::cereal::BinaryOutputArchive>::operator ()(bool,std::integer_sequence<size_t,Idx,Tail...>)': could not deduce template argument for '__formal'
1>        with
1>        [
1>            Predicate=tim::concepts::is_output_archive
1>        ]
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(426): message : see declaration of 'validate<Predicate,tim::cereal::PrettyJSONOutputArchive,tim::cereal::MinimalJSONOutputArchive,tim::cereal::XMLOutputArchive,tim::cereal::PortableBinaryOutputArchive,tim::cereal::BinaryOutputArchive>::operator ()'
1>        with
1>        [
1>            Predicate=tim::concepts::is_output_archive
1>        ]
1>Done building project "traits_tests.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

The errors without tim::enable_if are:

```
Build started...
1>------ Build started: Project: traits_tests, Configuration: Release x64 ------
1>traits_tests.cpp
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(405,1): error C2872: 'enable_if_t': ambiguous symbol
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\timemory/utility/types.hpp(179,56): message : could be 'tim::enable_if_t = std::enable_if<_Test,_Ty>::type'
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\timemory/components/types.hpp(58,56): message : or       'tim::component::enable_if_t = std::enable_if<_Test,_Ty>::type'
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(436): message : see reference to class template instantiation 'validate<Predicate,tim::component::cpu_clock,tim::component::cpu_util,tim::component::current_peak_rss,tim::component::data_tracker<intmax_t,tim::project::timemory>,tim::component::data_tracker<size_t,tim::project::timemory>,tim::component::data_tracker<double,tim::project::timemory>,tim::component::monotonic_clock,tim::component::monotonic_raw_clock,tim::component::page_rss,tim::component::peak_rss,tim::component::priority_context_switch,tim::component::process_cpu_clock,tim::component::process_cpu_util,tim::component::read_bytes,tim::component::system_clock,tim::component::thread_cpu_clock,tim::component::thread_cpu_util,tim::component::trip_count,tim::component::user_clock,tim::component::user_bundle<10000,tim::project::timemory>,tim::component::user_bundle<0,tim::project::kokkosp>,tim::component::user_bundle<22000,tim::project::timemory>,tim::component::user_bundle<20000,tim::project::timemory>,tim::component::voluntary_context_switch,tim::component::vtune_event,tim::component::vtune_frame,tim::component::vtune_profiler,tim::component::wall_clock,tim::component::written_bytes>' being compiled
1>        with
1>        [
1>            Predicate=tim::trait::is_available
1>        ]
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(488): message : see reference to class template instantiation 'validate<tim::trait::is_available,tim::available_types_t>' being compiled
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\tests\traits_tests.cpp(424,1): error C2872: 'enable_if_t': ambiguous symbol
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\timemory/utility/types.hpp(179,56): message : could be 'tim::enable_if_t = std::enable_if<_Test,_Ty>::type'
1>C:\Users\e27290\projects\crow-mfw\3rdparty\timemory\source\timemory/components/types.hpp(58,56): message : or       'tim::component::enable_if_t = std::enable_if<_Test,_Ty>::type'
1>Done building project "traits_tests.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```